Sources/AblyChat/RoomLifecycleManager.swift

```diff
@@ -11,31 +17,90 @@ internal protocol RoomLifecycleContributorChannel: Sendable {
     var errorReason: ARTErrorInfo? { get async }
+
+    /// Equivalent to subscribing to a `RealtimeChannelProtocol` object’s state changes via its `on(_:)` method. The subscription should use the ``BufferingPolicy.unbounded`` buffering policy.
+    ///
+    /// It is marked as `async` purely to make it easier to write mocks for this method (i.e. to use an actor as a mock).
+    func subscribeToState() async -> Subscription<ARTChannelStateChange>
 }
 
-internal actor RoomLifecycleManager<Channel: RoomLifecycleContributorChannel> {
-    /// A realtime channel that contributes to the room lifecycle.
-    internal struct Contributor {
-        /// The room feature that this contributor corresponds to. Used only for choosing which error to throw when a contributor operation fails.
-        internal var feature: RoomFeature
+/// A realtime channel that contributes to the room lifecycle.
+///
+/// The identity implied by the `Identifiable` conformance must distinguish each of the contributors passed to a given ``RoomLifecycleManager`` instance.
+internal protocol RoomLifecycleContributor: Identifiable, Sendable {
+    associatedtype Channel: RoomLifecycleContributorChannel
+
+    /// The room feature that this contributor corresponds to. Used only for choosing which error to throw when a contributor operation fails.
+    var feature: RoomFeature { get }
+    var channel: Channel { get }
+
+    /// Informs the contributor that there has been a break in channel continuity, which it should inform library users about.
+    ///
+    /// It is marked as `async` purely to make it easier to write mocks for this method (i.e. to use an actor as a mock).
+    func emitDiscontinuity(_ error: ARTErrorInfo) async
+}
 
-        internal var channel: Channel
+internal actor RoomLifecycleManager<Contributor: RoomLifecycleContributor> {
+    /// Stores manager state relating to a given contributor.
+    private struct ContributorAnnotation {
+        // TODO: Not clear whether there can be multiple or just one (asked in https://github.com/ably/specification/pull/200/files#r1781927850)
+        var pendingDiscontinuityEvents: [ARTErrorInfo] = []
     }
 
     internal private(set) var current: RoomLifecycle
     internal private(set) var error: ARTErrorInfo?
+    // TODO: This currently allows the the tests to inject a value in order to test the spec points that are predicated on whether “a channel lifecycle operation is in progress”. In https://github.com/ably-labs/ably-chat-swift/issues/52 we’ll set this property based on whether there actually is a lifecycle operation in progress.
+    private let hasOperationInProgress: Bool
+    /// Manager state that relates to individual contributors, keyed by contributors’ ``Contributor.id``. Stored separately from ``contributors`` so that the latter can be a `let`, to make it clear that the contributors remain fixed for the lifetime of the manager.
+    private var contributorAnnotations: ContributorAnnotations
```

Comment:
This could be just a dictionary [Contributor.ID: ContributorAnnotation], no?

Reply:
I created a type so that we don't have to force-unwrap or `guard`; the `ContributorAnnotations` type guarantees that subscripting will return non-`nil` for all of the contributors the manager is initialized with.

Reply:
I mean, force-unwrapping isn't the end of the world; I just thought this is maybe a bit neater.